servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NonPipelinedCloseHandler.java

```diff
@@ -162,6 +163,18 @@ void gracefulUserClosing(final Channel channel) {
         }
     }
 
+    @Override
+    void channelClose(final Channel channel) {
+        // We need to set SO_LINGER before we invoke close if we are not writing and still reading. This forces sending
+        // RST (instead of FIN) which the peer will interpret as "full close" and any future writes by the peer will
+        // fail (if the peer is half closed, not reading, but still writing). Otherwise, the peer write attempts may
+        // be accepted by the OS, only to generate RST when the local peer receives the data, but the peer has finished
+        // reading which may prevent processing the RST and be hung.
+        if (isAllSet(state, READ) && !isAllSet(state, WRITE) && channel instanceof SocketChannel) {
+            setSocketResetOnClose((SocketChannel) channel);
```

Comment:
NonPipelinedCloseHandler is used by child channels that represent a virtual stream. Does setting a socket option here make any difference?

Reply:
No `Http2StreamChannel` currently extends `Channel` currently (not `SocketChannel`). However if/when it does I don't think this will meaningfully change behavior.